smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/OpenApiJsonSchemaMapper.java

```diff
@@ -100,12 +94,38 @@ public Schema.Builder updateSchema(Shape shape, Schema.Builder builder, JsonSche
             }
         }
 
+        if (shape.hasTrait(BoxTrait.class)) {
+            handleNullability(builder, config);
+        }
+
         // Remove unsupported JSON Schema keywords.
         UNSUPPORTED_KEYWORD_DIRECTIVES.forEach(builder::disableProperty);
 
         return builder;
     }
 
+    private void handleFormatKeyword(Schema.Builder builder, OpenApiConfig config) {
+        String blobFormat = config.getDefaultBlobFormat();
+        if (config.getVersion() != null && config.getVersion().equals("3.1.0")) {
```

Comment:
by providing a default version and ensuring it's never set to null, you can remove the null check here

Reply:
OpenAPI will likely come out with more versions, and then we'll need to update any conditional checks for these. It might be a better approach for longer term maintainability to make the OpenAPI version an enum and then to add methods on it to describe what functionality it supports -- for example, `config.getVersion().supportsContentEncoding()` would be great instead of this, and it would automatically account for new versions.

Reply:
added `OpenApiVersion` enum, with methods `supportsNullability` and `usesFormatKeywordForFilePayload` and modified code as per the comment. in the future, for a new OpenAPI version, you won't have to change the conditionals, but just add a new enum value with correct booleans for whether they support `nullability` or not, and whether they use `format` keyword for file payloads.